Makefile

```diff
@@ -27,12 +27,17 @@ LD_FLAGS := "-w $(shell EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) $(REPO_ROOT)/vend
 # Rules for local development scenarios #
 #########################################
 
+ZAP_DEVEL     := true
+ZAP_LOG_LEVEL := debug
+
 .PHONY: start
 start:
 	@GO111MODULE=on go run \
 	    -mod=vendor \
 		-ldflags $(LD_FLAGS) \
 		./cmd/gardener-resource-manager \
+	  --zap-devel=$(ZAP_DEVEL) \
+	  --zap-log-level=$(ZAP_LOG_LEVEL) \
```

Comment:
I know that this isn't your implementation but I somehow don't like the inconsistency between named log levels here and `int` levels [here](https://github.com/gardener/gardener-resource-manager/pull/98/files#diff-107e32ddfe5be7868d4f9af43c0f4b309936b3856dba6ed11925cdb84f7dbd03R119).
1. The user doesn't know which log levels exist (could be part of the cmd help)
2. As a developer using `log.V(1).Info` or `log.Info` it's unclear at which log level which entry is printed.

Do you think it makes sense to either (1) mask the `--zap-log-level` and use `--v` with an Integer which maps back to a zap-log-level log level or (2) use a constant instead of Integers in the code, e.g. `log.V(1).Info --> log.V(Debug).Info`.

Reply:
Discussed this with @timuthy on the phone:
We both find the current implementation to be a bit confusing, especially the stuff with the level inversion that is going on in the zap options in c-r and in zapr.

We think it makes sense to check, how other logging implementations are handling the log levels, e.g. the new klog implementation.
It probably makes sense to add own flags (like `--v`) to a) be flexible to change the underlying logging library one day without breaking compatibility and b) make it more aligned with kubernetes components.

In general, it might also make sense to do similar refactorings in the extensions and also gardener itself and have everything consistent.
So let's look into this a bit, so we can have a bit of a guideline for this later on, before we end up with a zoo of different ways to handle logging in our components :D 

/author-action
/needs changes